### PR TITLE
Updating admin sdk to tsc compiler

### PR DESCRIPTION
**Pull request by vafl-brut**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+"use strict";
+exports.__esModule = true;
+function sayHi() {
+    console.log('hi');
```
Comment (snapjay) [inline src/index.js]: Hi :)

File: src/Podium/Request.ts [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
+                ) || error.response
+            return {
+                code: error.response.data.code,
+                data: error.response.data as IResponse,
+                message,
+                status: error.response.status,
+                statusText: error.response.statusText,
+            }
+        } else {
+            // todo: fix this
```
Comment (snapjay) [inline src/Podium/Request.ts]: What is wrong with this?

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+}
+exports.sayHi = sayHi;
+function addNumbers(a, b) {
+    return a + b;
+}
+exports.addNumbers = addNumbers;
+//
```
Comment (snapjay) [inline src/index.js]: This too

File: tsconfig.json [hunk 4]
```diff
@@ -1,12 +1,67 @@
...
-    "module": "commonjs",
-    "target": "es6"
+    /* Basic Options */
+    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    "lib": [
+      "es2017",
+      "dom"
+    ],                             /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
```
Comment (snapjay) [inline tsconfig.json]: I like the idea of having all these here

File: tsconfig.json [hunk 5]
```diff
@@ -1,12 +1,67 @@
...
+    // "composite": true,                     /* Enable project compilation */
+    // "removeComments": true,                /* Do not emit comments to output. */
+    // "noEmit": true,                        /* Do not emit outputs. */
+    // "importHelpers": true,                 /* Import emit helpers from 'tslib'. */
+    // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
+    // "isolatedModules": true,               /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
+
+    /* Strict Type-Checking Options */
+    "strict": true,                           /* Enable all strict type-checking options. */
+    // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
```
Comment (snapjay) [inline tsconfig.json]:
```suggestion
    "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
```

File: tsconfig.json [hunk 6]
```diff
@@ -1,12 +1,67 @@
...
+    "lib": [
+      "es2017",
+      "dom"
+    ],                             /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
+    // "checkJs": true,                       /* Report errors in .js files. */
+    // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    // "declarationMap": true,                /* Generates a sourcemap for each corresponding '.d.ts' file. */
+    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
```
Comment (snapjay) [inline tsconfig.json]:
```suggestion
    "sourceMap": true,                     /* Generates corresponding '.map' file. */
```

File: tslint.json [hunk 7]
```diff
@@ -5,15 +5,31 @@
...
+            true,
+            "never"
+        ],
+        "typedef": [
+            true,
+            "call-signature",
+            "parameter",
+            "member-variable-declaration"
+        ],
         "no-console": false,
```
Comment (snapjay) [inline tslint.json]:
```suggestion
        "no-console": true,
```

File: tslint.json [hunk 8]
```diff
@@ -5,15 +5,31 @@
...
+            {
+                "import-sources-order": "any",
+                "named-imports-order": "case-insensitive"
+            }
+        ]
     },
     "rulesDirectory": []
-}
\ No newline at end of file
+}
```
Comment (snapjay) [inline tslint.json]:
```suggestion
}

```

File: types/index.d.ts [hunk 9]
```diff
@@ -9,13 +34,15 @@ export interface IPodiumError {
...
+    code: string
+    message: string
     data: IResponse
     status: number
     statusText: string
 }
 
 export interface IPodiumPromise<T> extends Promise<T> {
-    finally?: string
+    // finally?: () => void | null
```
Comment (snapjay) [inline types/index.d.ts]: Should this be commented?

File: src/index.js (hunk 1 above)
Comment (vafl-brut) [reply src/index.js]: Hi :)

File: src/Podium/Request.ts (hunk 2 above)
Comment (vafl-brut) [reply src/Podium/Request.ts]: So AxiosError might not have .response, so i had to add if and else part i came up with randomly

File: types/index.d.ts (hunk 9 above)
Comment (vafl-brut) [reply types/index.d.ts]: no sure. copied it over like this

File: src/Podium/Request.ts (hunk 2 above)
Comment (snapjay) [reply src/Podium/Request.ts]: Can you  at lease add that to the comment;

File: src/index.js (hunk 1 above)
Comment (snapjay) [reply src/index.js]: Apart from the whimsical friendliness of this function; do we need it?
(also addNumbers and addValues)

Comment (snapjay) [reply src/index.js]: Apart from the whimsical friendliness of this function; do we need this?

Comment (vafl-brut) [reply src/index.js]: no, that whole file should be gone now :)

**Pull request merged**